cashu/lightning/lnmarkets.py

```diff
@@ -0,0 +1,349 @@
+        if self.unit == Unit.usd:
+            # We do this trick to avoid messing up the signature.
+            amount_usd = float(amount.to_float_string())
+            amount_usd = (
+                int(amount_usd) if float(int(amount_usd)) == amount_usd else amount_usd
+            )
+            data = {"amount": amount_usd, "currency": "usd"}
+            path = "/v2/user/deposit/susd"
+        else:
+            data = {"amount": amount.amount}
+            path = "/v2/user/deposit"
+
+        logger.debug(f"{data = } {path = }")
+        assert data and path
+        headers = await self.get_request_headers(Method.POST, path, data)
+        try:
+            r = await self.client.post(
+                url=f"{self.endpoint}{path}",
+                json=data,
+                headers=headers,
+            )
+            raise_if_err(r)
+        except CashuError as e:
+            return InvoiceResponse(
+                ok=False,
+                error_message=f"Error while creating invoice: {e.detail}",
+            )
+
+        data = None
+        try:
+            data = r.json()
+        except Exception:
+            logger.error(f"Received invalid response from {self.endpoint}: {r.text}")
+            return InvoiceResponse(
+                ok=False,
+                error_message=(
+                    f"Received invalid response from {self.endpoint}: {r.text}"
+                ),
+            )
+
+        checking_id, payment_request = data["depositId"], data["paymentRequest"]
+        assert isinstance(checking_id, str) and isinstance(payment_request, str)
+
+        return InvoiceResponse(
+            ok=True,
+            checking_id=checking_id,
+            payment_request=payment_request,
+        )
+
+    async def pay_invoice(
+        self, quote: MeltQuote, fee_limit_msat: int
+    ) -> PaymentResponse:
+        self.assert_unit_supported(Unit[quote.unit])
+
+        data = {"invoice": quote.request}
+        path = "/v2/user/withdraw"
+        if self.unit == Unit.usd:
+            data["quote_id"] = quote.checking_id
+
+        headers = await self.get_request_headers(Method.POST, path, data)
+        try:
+            r = await self.client.post(
+                url=f"{self.endpoint}{path}",
+                json=data,
+                headers=headers,
+                timeout=None,
+            )
+            raise_if_err(r)
+        except CashuError as e:
+            return PaymentResponse(error_message=f"payment failed: {e.detail}")
+
+        try:
+            data = r.json()
+        except Exception:
+            logger.error(f"payment failed: {r.text}")
+            return PaymentResponse(error_message=f"payment failed: {r.text}")
+
+        # lnmarkets does not provide a payment_preimage :(
+        checking_id = data["id"]
+        return PaymentResponse(
+            ok=True,
+            checking_id=checking_id,
+            fee=Amount(unit=self.unit, amount=quote.fee_reserve),
+        )
```

Comment:
We need to return the actually paid fee here instead of the fee_reserve (otherwise there won't be any fee refunds for overpaid fees).